### ISSUE TITLE
Android 8 compatibility with NotificationExtenderService when targeting API Level 26

### ISSUE THREAD

**Issue by Reza-Soltani** [OneSignal/OneSignal-Android-SDK#332]
My app works right when my target is 25 or less, and I am getting push notifications.
But when I changed target sdk to 26 and testing my app on API 26 image with google play in emulator, when my app is in background and it receives a new push notifications, my app will crash.
Do you know what’s the problem?

```Java
10-14 13:26:26.201 28411-28411/? E/AndroidRuntime: FATAL EXCEPTION: main
Process: ir.divar, PID: 28411
   java.lang.RuntimeException: Unable to start receiver com.onesignal.GcmBroadcastReceiver: java.lang.IllegalStateException: Not allowed to start service Intent { act=com.onesignal.NotificationExtender pkg=ir.divar (has extras) }: app is in background uid UidRecord{496b8b2 u0a82 RCVR idle procs:1 seq(0,0,0)}
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:3259)
       at android.app.ActivityThread.-wrap17(Unknown Source:0)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1677)
       at android.os.Handler.dispatchMessage(Handler.java:105)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6540)
       at java.lang.reflect.Method.invoke(Native Method)
       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
    Caused by: java.lang.IllegalStateException: Not allowed to start service Intent { act=com.onesignal.NotificationExtender pkg=ir.divar (has extras) }: app is in background uid UidRecord{496b8b2 u0a82 RCVR idle procs:1 seq(0,0,0)}
       at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1497)
       at android.app.ContextImpl.startService(ContextImpl.java:1453)
       at android.content.ContextWrapper.startService(ContextWrapper.java:644)
       at android.content.ContextWrapper.startService(ContextWrapper.java:644)
       at android.support.v4.content.WakefulBroadcastReceiver.startWakefulService(WakefulBroadcastReceiver.java:99)
       at com.onesignal.NotificationBundleProcessor.processBundle(NotificationBundleProcessor.java:367)
       at com.onesignal.GcmBroadcastReceiver.processOrderBroadcast(GcmBroadcastReceiver.java:69)
       at com.onesignal.GcmBroadcastReceiver.onReceive(GcmBroadcastReceiver.java:59)
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:3252)
       at android.app.ActivityThread.-wrap17(Unknown Source:0) 
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1677) 
       at android.os.Handler.dispatchMessage(Handler.java:105) 
       at android.os.Looper.loop(Looper.java:164) 
       at android.app.ActivityThread.main(ActivityThread.java:6540) 
       at java.lang.reflect.Method.invoke(Native Method) 
       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240) 
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767) 
```

**Comment by jkasten2** [comment]
@Reza-Soltani From the strack trace it looks you are using the OneSignal SDK version 3.5.3 or older. Can you update to the latest 3.6.3 version? We recommend using the following in your `build.gradle` to make sure you stay up to date.
```groovy
compile 'com.onesignal:OneSignal:[3.6.2, 3.99.99]'
```

**Comment by Reza-Soltani** [comment]
You were right, I was using 3.5.1 version.
But when I upgraded my version to 3.6.3, then no notification will be showed. without any error.

**3.6.3:**
**Case that app is in background:** Nothing happened, onNotificationProcessing of MyNotificationExtenderService didn't called.
**Case that app is in foreground:** MyNotificationExtenderService will be called, then I will call displayNotification, but I can’t see any notification on the phone.

When the version was 3.5.1, at least I had foreground notifications.

Btw, thank you

**Comment by Reza-Soltani** [comment]
I fixed the foreground situation and now it work correctly.
But I still have problem with background case.

Only related log in logcat
10-16 12:27:54.242 1529-1584/? W/ActivityManager: Background start not allowed: service Intent { act=com.onesignal.NotificationExtender flg=0x4 pkg=ir.divar (has extras) } to ir.divar/.notification.DivarNotificationExtender from pid=-1 uid=10083 pkg=ir.divar

**Comment by mywaystar** [comment]
The NotificationExtender is using IntentService which is deprecated starting Android O in favor on JobIntentService

I'll create a fork today.

**Comment by jkasten2** [comment]
Unfortunately using a `JobIntentService` would require app projects to be upgraded to `compileSdkVersion 26` which will won't be possible for everyone to upgrade to yet. We are researching strategies on resolving this and will update this issue when we have more details.

**Comment by jkasten2** [comment]
This is now fixed in release 3.6.5. Please see the [release notes](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/3.6.5) for the required changes.
